src/App.tsx

```diff
@@ -1,223 +1,197 @@
+import React, {
+  useContext, useEffect, useRef, useState, useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Error } from './components/Error';
+import { Filter } from './components/Filter';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortFilter, setSortFilter] = useState('all');
+  const [visibelTodos, setVisibelTodos] = useState<Todo[]>([...todos]);
+  const [completedTodos, setCompletedTodos]
+  = useState<Todo[]>([...todos].filter(todo => todo.completed));
+
+  const [isError, setError] = useState(false);
+  const [messageError, setMessageError] = useState('');
+  const [newTodoTitle, setTitle] = useState('');
+  const [activeTodoId, setActiveTodoId] = useState(0);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(response => {
+          setTodos(response);
+        })
+        .catch(() => {
+          setError(true);
+          setMessageError('Todos from server were not gotten');
+        });
+    }
   }, []);
 
+  useMemo(() => {
+    setVisibelTodos(() => (
+      todos.filter(todo => {
+        switch (sortFilter) {
+          case 'active':
+            return !todo.completed;
+
+          case 'completed':
+            return todo.completed;
+
+          default:
+            return true;
+        }
+      })));
+  }, [todos, sortFilter]);
+
+  useMemo(() => {
+    setCompletedTodos(() => todos.filter(todo => todo.completed));
+  }, [todos]);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!user) {
+      return;
+    }
+
+    if (newTodoTitle.trim().length === 0) {
+      setError(true);
+      setMessageError('Title can\'t be empty');
+
+      return;
+    }
+
+    setVisibelTodos(state => [...state, {
+      id: 0,
+      userId: user.id,
+      title: newTodoTitle,
+      completed: false,
+    }]);
+
+    try {
+      const isAdding = await createTodo({
+        userId: user.id,
+        title: newTodoTitle,
+        completed: false,
+      });
+
+      setTodos(state => [...state, isAdding]);
+    } catch (errorFromServer) {
+      setError(true);
+      setMessageError('Unable to add a todo');
+    }
+
+    setTitle('');
+  };
+
+  const handleRemoveTodo = async (removeTodoID: number) => {
+    setActiveTodoId(removeTodoID);
+
+    try {
+      await deleteTodo(removeTodoID);
+
+      setTodos(state => state.filter(todo => todo.id !== removeTodoID));
```

Comment:
```suggestion
      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== removeTodoID));
```
A variable name should describe what is stored in it.

Reply:
done